docs/docs/higher-order-components.md

```diff
@@ -0,0 +1,488 @@
+      super();
+      this.handleChange = this.handleChange.bind(this);
+      this.state = {
+        data: selectData(DataSource, props)
+      };
+    }
+
+    componentDidMount() {
+      // ... that takes care of the subscription...
+      DataSource.addChangeListener(this.handleChange);
+    }
+
+    componentWillUnmount() {
+      DataSource.removeChangeListener(this.handleChange);
+    }
+
+    handleChange() {
+      this.setState({
+        comments: selectData(DataSource, this.props)
+      });
+    }
+
+    render: function() {
+      // ... and renders the wrapped component with the fresh data!
+      // Notice that we pass through any additional props
+      return <WrappedComponent data={this.state.data} {...this.props} />;
+    }
+  });
+}
+```
+
+And that's it! The wrapped component receives a single prop, `data`, which it uses to render its output. The HOC isn't concerned with how or why the data is used, and the wrapped component isn't concerned with where the data came from.
+
+Because `withSubscription` is a normal function, you can add as many or as few arguments as you like. For example, you may want to make the name of the `data` prop configurable, to further isolate the HOC from the wrapped component. Or you could accept an argument that configures `shouldComponentUpdate`, or one that configures the data source. These are all possible because the HOC has full control over how the component is defined.
+
+Note that like components, the contract between `withDataSubscription` and the wrapped component is entirely props-based. This makes it easy to swap one HOC for a different one, as long as they provide the same props to the wrapped component. An example of how this may be useful is if you change data-fetching libraries.
+
+### Why Not Use a Component?
+
+Components are a good choice when you want to vary the output based on the given props. However, in our example, the difference between `CommentList` and `BlogPost` isn't limited to their output. They also differ in the implementation of their lifecycle methods.
+
+Because HOCs control the definition of the component class itself, they have the power to configure all aspects of a component's behavior, not just what goes in the `render` method.
+
+>**Note:**
+>
+> It's possible to use normal components in place of certain types of HOCs. See the document on render callbacks to learn more about this technique and how it compares.
+
+## Conventions for Idiomatic Higher-Order Components
+
+Although HOCs are a pattern rather than a first-class part of React's API, there are some conventions that we recommend you follow.
+
+### Don't Mutate the Original Component. Use Composition.
+
+Resist the temptation to modify a component's prototype (or otherwise mutate it) inside an HOC.
+
+```js
+function onlyUpdateForKeys(InputComponent, keys) {
+  // Mutates the input component's prototype
+  // *Don't do this!*
+  InputComponent.prototype.shouldComponentUpdate = function(nextProps) {
```

Comment:
Can we use another example? AFAIK we want to treat `sCU` less strictly in the future and it would be nice if we didn't give an impression it can be legitimately used for e.g. blocking UI. Maybe data fetching would work better here?